mlir/unittests/translation/test_translation.cpp

```diff
@@ -739,6 +740,49 @@ TEST_F(ImportTest, IfElseRegister) {
   ASSERT_TRUE(checkOutput(checkString, outputString));
 }
 
+TEST_F(ImportTest, IfElseHandlingFromQasm) {
+  const qc::QuantumComputation qc = qasm3::Importer::imports("OPENQASM 3.0;"
+                                                             "qubit q;"
+                                                             "bit c;"
+                                                             "h q;"
+                                                             "c = measure q;"
+                                                             "if(c) {"
+                                                             "  x q;"
+                                                             "} else {"
+                                                             "  h q;"
+                                                             "}"
+                                                             "c = measure q;");
```

Comment:
This has nothing to do with your PR in particular, but I find it questionable that this would produce compound operations, given how every branch is only a single operation. 
I'd actually consider that a missed opportunity for optimization or even a bug.
Two things:
- would you mind adjusting the QASM parser to produce single operations when possible?
- would you mind using a test program here that very clearly uses more than one operation in one of the branches?